nmpolicy/internal/capture/capture.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of the nmpolicy project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package capture
+
+import (
+	"fmt"
+
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/ast"
+	"github.com/nmstate/nmpolicy/nmpolicy/internal/lexer"
+	"github.com/nmstate/nmpolicy/nmpolicy/types"
+)
+
+type CapsState = map[string]types.CaptureState
```

Comment:
Can we move all the types to `types.go` ?

What do you think about moving the interfaces to their packages and unexporting the structs ? 


```golang
package lexer

type Lexer interface ...
type lexer struct ...
```

Reply:
This type alias is a leftover, per your request I removed them all.

Interfaces are to be defined either at the usage or in a separate package, they should almost never be defined in the implementation package.

Also, note that `types` represent public API types, and these probably do not fit there.